telegram/base.py

```diff
@@ -17,36 +17,64 @@
 # You should have received a copy of the GNU Lesser Public License
 # along with this program.  If not, see [http://www.gnu.org/licenses/].
 """Base class for Telegram Objects."""
-
 try:
     import ujson as json
 except ImportError:
-    import json
+    import json  # type: ignore[no-redef]
 
 import warnings
 
+from telegram.utils.types import JSONDict
+from typing import Tuple, Any, Optional, Type, TypeVar, TYPE_CHECKING, List
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+TO = TypeVar('TO', bound='TelegramObject')
```

Comment:
covariant=True?

Reply:
I'm not too familiar with the details, but from what I understand I guess you're right. E.g. in any place, where `Message` is valid, any subclass of `Message` should be valid